pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -0,0 +1,1577 @@
+
+			let currency_length = currencies.len();
+
+			let current_asset_id = NextAssetId::<T>::get();
+
+			let (currency_ids, next_asset_id) = Self::generate_sequential_asset_ids(current_asset_id, currency_length)?;
+
+			let pool_id = T::PoolId::from(currency_ids.blake2_256());
+
+			// Todo: change that.
+			T::DepositCurrency::hold(
+				&T::RuntimeHoldReason::from(HoldReason::Deposit),
+				&who,
+				Self::calculate_pool_deposit(currency_length),
+			)?;
+
+			let pool_account = &pool_id.clone().into();
+
+			// Touch the pool account in order to be able to transfer the collateral
+			// currency to it. This should also verify that the currency actually exists.
+			T::CollateralCurrencies::touch(collateral_id.clone(), pool_account, &who)?;
+
+			currencies
+				.into_iter()
+				.zip(currency_ids.iter())
+				.try_for_each(|(entry, asset_id)| -> DispatchResult {
+					let TokenMeta {
+						min_balance,
+						name,
+						symbol,
+					} = entry;
+
+					T::Fungibles::create(asset_id.clone(), pool_account.to_owned(), false, min_balance)?;
+
+					// set metadata for new asset class
+					T::Fungibles::set(
+						asset_id.to_owned(),
+						pool_account,
+						name.into_inner(),
+						symbol.into_inner(),
+						denomination,
+					)?;
+
+					Ok(())
+				})?;
+
+			Pools::<T>::set(
+				&pool_id,
+				Some(PoolDetails::new(
+					who,
+					checked_curve,
+					collateral_id,
+					currency_ids,
+					transferable,
+					denomination,
+					min_operation_balance,
+				)),
+			);
+
+			// update the storage for the next tx.
+			NextAssetId::<T>::set(next_asset_id);
+
+			Self::deposit_event(Event::PoolCreated { id: pool_id });
+
+			Ok(())
+		}
+
+		/// Resets the managing team of a pool. The new team will be set to the
+		/// provided team. The currency index is used to select the currency
+		/// that the team will manage.
+		/// The origin account must be a manager of the pool.
+		///
+		/// # Parameters
+		/// - `origin`: The origin of the call, requiring the caller to be a
+		///   manager of the pool.
+		/// - `pool_id`: The identifier of the pool.
+		/// - `team`: The new managing team.
+		/// - `currency_idx`: The index of the currency in the bonded currencies
+		///   vector.
+		///
+		/// # Returns
+		/// - `DispatchResult`: The result of the dispatch.
+		///
+		/// # Errors
+		/// - `Error::<T>::PoolUnknown`: If the pool does not exist.
+		/// - `Error::<T>::NoPermission`: If the caller is not a manager of the
+		///   pool.
+		/// - `Error::<T>::IndexOutOfBounds`: If the currency index is out of
+		///   bounds.
+		/// - Other errors depending on the types in the config.
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::reset_team())]
+		pub fn reset_team(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			team: PoolManagingTeam<AccountIdOf<T>>,
+			currency_idx: u32,
+		) -> DispatchResult {
+			let who = T::DefaultOrigin::ensure_origin(origin)?;
```

Comment:
On a locked pool minting and burning is a management operation, and I don't see why these would not be done from an xcm origin. What's the technical reason that we need different origins anyway? Can they not be configured to allow either a signed or an xcm origin? Can xcm not be configured to call extrinsic that would otherwise require a signed origin? I don't understand enough of xcm to see what your vision is with these origins.